helix-view/src/handlers/lsp.rs

```diff
@@ -214,29 +220,36 @@ impl Editor {
         Ok(())
     }
 
-    fn apply_document_resource_op(&mut self, op: &lsp::ResourceOp) -> std::io::Result<()> {
+    fn apply_document_resource_op(
+        &mut self,
+        op: &lsp::ResourceOp,
+    ) -> Result<(), ApplyEditErrorKind> {
+        fn wrap_io(err: std::io::Error) -> ApplyEditErrorKind {
+            ApplyEditErrorKind::IoError(err)
+        }
```

Comment:
this should not exist. We should implement `From<io::Error> for ApplyEditErrorKind` then `?` would handle the conversion automatically